Adapt celmatcher.go for google/cel-go v0.29.x API changes.

Caddy v2.11.4 still pins cel-go v0.28.1; interpreter.NewCall now expects
[]interpreter.InterpretableV2. Upstream merged the same fix on master in
https://github.com/caddyserver/caddy/pull/7872 (2026-07-11).

Upstream PR: https://github.com/caddyserver/caddy/pull/7872
Advisory: https://github.com/advisories/GHSA-gcjh-h69q-9w9g

diff --git a/modules/caddyhttp/celmatcher.go b/modules/caddyhttp/celmatcher.go
--- a/modules/caddyhttp/celmatcher.go
+++ b/modules/caddyhttp/celmatcher.go
@@ -503,7 +503,7 @@ func CELMatcherDecorator(funcName string, fac any) interpreter.InterpretableDeco
 			}
 			return interpreter.NewCall(
 				i.ID(), funcName, funcName+"_opt",
-				[]interpreter.Interpretable{reqAttr},
+				[]interpreter.InterpretableV2{reqAttr},
 				func(args ...ref.Val) ref.Val {
 					// The request value, guaranteed to be of type celHTTPRequest
 					celReq := args[0]
@@ -526,7 +526,7 @@ func CELMatcherDecorator(funcName string, fac any) interpreter.InterpretableDeco
 			}
 			return interpreter.NewCall(
 				i.ID(), funcName, funcName+"_opt",
-				[]interpreter.Interpretable{reqAttr},
+				[]interpreter.InterpretableV2{reqAttr},
 				func(args ...ref.Val) ref.Val {
 					// The request value, guaranteed to be of type celHTTPRequest
 					celReq := args[0]
